docs/book/user-guide/tutorial/trigger-pipelines-from-external-systems.md

```diff
@@ -0,0 +1,879 @@
+    # For demonstration, we'll create synthetic data
+    np.random.seed(42)
+    n_samples = 1000
+    
+    print(f"Loading data from: {data_url}")
+    # In a real scenario, you'd load from data_url
+    # E.g., pd.read_csv(data_url)
+    
+    data = pd.DataFrame({
+        "feature_1": np.random.normal(0, 1, n_samples),
+        "feature_2": np.random.normal(0, 1, n_samples),
+        "feature_3": np.random.normal(0, 1, n_samples),
+        "target": np.random.choice([0, 1], n_samples),
+    })
+    return data
+
+@step
+def preprocess(data: pd.DataFrame) -> Dict[str, Any]:
+    """Split data into train and test sets."""
+    X = data.drop("target", axis=1)
+    y = data["target"]
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=0.2, random_state=42
+    )
+    return {
+        "X_train": X_train,
+        "X_test": X_test,
+        "y_train": y_train,
+        "y_test": y_test,
+    }
+
+@step
+def train_model(
+    datasets: Dict[str, Any], model_type: str = "random_forest"
+) -> Union[RandomForestClassifier, GradientBoostingClassifier]:
+    """Train a model based on the specified type."""
+    X_train = datasets["X_train"]
+    y_train = datasets["y_train"]
+    
+    if model_type == "random_forest":
+        model = RandomForestClassifier(n_estimators=100, random_state=42)
+    elif model_type == "gradient_boosting":
+        model = GradientBoostingClassifier(random_state=42)
+    else:
+        raise ValueError(f"Unknown model type: {model_type}")
+    
+    print(f"Training a {model_type} model...")
+    model.fit(X_train, y_train)
+    return model
+
+@step
+def evaluate(
+    datasets: Dict[str, Any],
+    model: Union[RandomForestClassifier, GradientBoostingClassifier],
+) -> Dict[str, float]:
+    """Evaluate the model and return metrics."""
+    X_test = datasets["X_test"]
+    y_test = datasets["y_test"]
+    
+    y_pred = model.predict(X_test)
+    accuracy = accuracy_score(y_test, y_pred)
+    
+    print(f"Model accuracy: {accuracy:.4f}")
+    return {"accuracy": float(accuracy)}
+
+# Define Docker settings for the pipeline
+docker_settings = DockerSettings(
+    requirements="requirements.txt",
+    required_integrations=["sklearn"],
+    python_package_installer="uv",
+)
+
+@pipeline(settings={"docker": docker_settings})
+def training_pipeline(
+    data_url: str = "example-data-source", model_type: str = "random_forest"
+):
+    """A configurable training pipeline that can be triggered externally."""
+    data = load_data(data_url)
+    datasets = preprocess(data)
+    model = train_model(datasets, model_type)
+    metrics = evaluate(datasets, model)
+    return metrics
+```
+
+### Creating a Requirements File
+
+Create a `requirements.txt` file with the necessary dependencies:
+
+```plaintext
+# Requirements for pipeline trigger API
+fastapi>=0.95.0
+uvicorn>=0.21.0
+requests>=2.28.0
+# Core dependencies
+scikit-learn>=1.0.0
+pandas>=1.3.0
+numpy>=1.20.0
+# ZenML
+zenml>=0.80.1
```

Comment:
they also need stack dependencies right?

Reply:
No that comes in the `Dockerfile`. You told me to do it that way. It's installed dynamically based on whatever the stack requires.